Tests/PactConsumerSwiftTests/AnimalServiceClient.swift

```diff
@@ -9,128 +8,162 @@ public struct Animal {
 }
 
 open class AnimalServiceClient {
-    fileprivate let baseUrl: String
+  fileprivate let baseUrl: String
+  private let session = URLSession(configuration: URLSessionConfiguration.ephemeral)
 
   public init(baseUrl : String) {
     self.baseUrl = baseUrl
   }
 
   open func getAlligators(_ success: @escaping (Array<Animal>) -> Void, failure: @escaping (NSError?) -> Void) {
-    Alamofire.request("\(baseUrl)/alligators")
-            .responseJSON {
-              (result) in
-              if result.result.isSuccess {
-                if let jsonArray = result.result.value as? Array<[String: AnyObject]> {
-                  success(jsonArray.map { animal -> Animal in
-                    return Animal(
-                            name: animal["name"] as! String,
-                            type: animal["type"] as! String,
-                            dob: animal["dateOfBirth"] as? String,
-                            legs: animal["legs"] as? Int)
-                  })
-                } else {
-                  failure(result.result.value as? NSError)
-                }
-              }
-            }
+    self.performRequest("\(baseUrl)/alligators") { jsonObject, nsError in
+      if let jsonArray = jsonObject as? Array<[String: AnyObject]> {
+        success(jsonArray.map { animal -> Animal in
+          return Animal(
```

Comment:
I understand this is a test, though would suggest setting up an `Animal` class that conforms to `Codable` protocol. Type casting would not be required anymore.

Reply:
I agree